pkg/provisioner/ironic/ironic.go

```diff
@@ -1760,34 +1761,60 @@ func (p *ironicProvisioner) RemoveBMCEventSubscriptionForNode(subscription metal
 	return operationComplete()
 }
 
-// TODO(hroyrh) : Replace with GetDataImageStatus function once the virtua_media.get
-// api is available.
-// Checks if the last VirtualMedia action(attach/detach) to a BareMetalHost was
-// successful of not.
-func (p *ironicProvisioner) IsDataImageReady() (isNodeBusy bool, nodeError error) {
-	// TODO(hroyrh)
-	// Get BareMetalHost VirtualMedia details using a GET api to vmedia
-
-	// Check if Ironic API version supports DataImage API
-	// Needs version >= 1.89
-	if !p.availableFeatures.HasDataImage() {
-		return true, fmt.Errorf("ironic version=%d doesn't support DataImage API, needs version>=1.89", p.availableFeatures.MaxVersion)
+// Uses the Ironic Virtual Media Get API which synchronously fetches the
+// virtual media details for the given node.
+// We return only the bool isImageAttached because the url in the response
+// body is always the same - based on the node uuid, so it is not useful
+// for any comparison purpose.
+func (p *ironicProvisioner) GetDataImageStatus() (isImageAttached bool, err error) {
+	// Check if Ironic API version supports Virtual Media Get API
+	// Needs version >= 1.93
+	if !p.availableFeatures.HasVirtualMediaGetAPI() {
```

Comment:
This is okay for a rarely used feature. If it was something more core, we'd need to gracefully fall back to the previous approach.

Reply:
Do you mean, in case of api version >= 1.89 and < 1.93, we should fall back to "IsDataImageReady" logic ?

Reply:
In theory. In practice, this feature is rarely used, and in OpenShift we couple BMO and Ironic versions, so I don't suggest we complicated the code further.